src/main/java/org/jboss/mjolnir/server/bean/UserRepositoryBean.java

```diff
@@ -92,84 +100,108 @@ public KerberosUser getOrCreateUser(String kerberosName) throws SQLException {
         return user;
     }
 
-    private void insertUser(KerberosUser user) throws SQLException {
-        final Connection connection = dataSource.getConnection();
-        try {
-            PreparedStatement statement;
-            statement = connection.prepareStatement(INSERT_USER_SQL);
-            statement.setString(1, user.getGithubName());
-            statement.setString(2, user.getName());
-            statement.setBoolean(3, user.isWhitelisted());
-            statement.executeUpdate();
-            statement.close();
-        } finally {
-            connection.close();
-        }
+    private void insertUser(KerberosUser user) {
+        UserEntity userEntity = convertUser(user);
+
+        Session session = sessionFactory.openSession();
+        session.beginTransaction();
+        session.save(userEntity);
+        session.getTransaction().commit();
+        session.close();
     }
 
-    private int updateUser(KerberosUser user) throws SQLException {
-        final Connection connection = dataSource.getConnection();
-        try {
-            PreparedStatement statement;
-            statement = connection.prepareStatement(UPDATE_USER_SQL);
-            statement.setString(1, user.getName());
-            statement.setString(2, user.getGithubName());
-            statement.setBoolean(3, user.isWhitelisted());
-            statement.setString(4, user.getName());
-            statement.setString(5, user.getGithubName());
-            final int affectedRows = statement.executeUpdate();
-            statement.close();
-            return affectedRows;
-        } finally {
-            connection.close();
+    private boolean updateUser(KerberosUser user) {
+        UserEntity userEntity = getUserFromDB(user);
+
+        if(userEntity == null) {
+            //user is not stored in the DB
+            return false;
         }
+
+        Session session = sessionFactory.openSession();
+        session.beginTransaction();
+        session.update(userEntity);
+        session.getTransaction().commit();
+        session.close();
+
+        return true;
```

Comment:
Theoretically it would be better load current user from database and update it inside a single transaction to avoid concurrency issues.